### PR TITLE
chore(warehouse): add tracking plan version when populating context violations

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Add tracking plan version when populating context violations

## Notion Ticket

https://www.notion.so/rudderstacks/Add-tracking-plan-version-Id-during-context-population-fdaa08e9936b4d409a81dc33f07b3200

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: processor/trackingplan.go [hunk 1]
```diff
@@ -84,7 +85,7 @@ func (proc *HandleT) validateEvents(groupedEventsByWriteKey map[WriteKeyT][]tran
 			continue
 		}
 
-		enhanceWithViolation(response, eventList[0].Metadata.TrackingPlanId)
+		enhanceWithViolation(response, eventList[0].Metadata.TrackingPlanId, eventList[0].Metadata.TrackingPlanVersion)
```
Comment (Jayachand) [inline processor/trackingplan.go:88]: Should we add unit tests for this func

**Review by Jayachand — APPROVED** [review]
LGTM

File: processor/trackingplan.go (hunk 1 above)
Comment (achettyiitr) [reply processor/trackingplan.go:88]: Good idea.

**Pull request merged**